packages/ui/src/components/token/AddTokenManualView.tsx

```diff
@@ -138,31 +139,29 @@ const AddTokenManualView = ({
         // eslint-disable-next-line react-hooks/exhaustive-deps
     }, [values.tokenSymbol, values.tokenAddress, setMessage])
 
-    const onAddressChange = (value: string) => {
+    const onAddressChange = async (value: string) => {
         if (!setSubmitEnabled) return
 
         setSubmitEnabled(false)
         if (utils.isAddress(value)) {
             setError("tokenAddress", { message: undefined })
             setIsLoading(true)
-            searchTokenInAssetsList(value)
-                .then((res) => {
-                    setIsLoading(false)
-                    if (res && res.length) {
-                        setValue("tokenAddress", res[0].address)
-                        setValue("tokenDecimals", res[0].decimals.toString())
-                        setValue("tokenLogo", res[0].logo)
-                        setValue("tokenName", res[0].name)
-                        setValue("tokenSymbol", res[0].symbol)
-                        setValue("tokenType", res[0].type)
+            const res = await searchTokenInAssetsList(value)
+            if (res) {
+                setIsLoading(false)
+                if (res && res.length) {
+                    setValue("tokenAddress", res[0].address)
+                    setValue("tokenDecimals", res[0].decimals.toString())
+                    setValue("tokenLogo", res[0].logo)
+                    setValue("tokenName", res[0].name)
+                    setValue("tokenSymbol", res[0].symbol)
+                    setValue("tokenType", res[0].type)
 
-                        setSubmitEnabled(true)
-                    }
-                })
-                .catch((err) => {
-                    setIsLoading(false)
-                    console.log("ERR: ", err)
-                })
+                    setSubmitEnabled(true)
+                }
+            } else {
```

Comment:
You don't need this else clause since the loading will be turned off in both if branches. 

```javascript
setIsLoading(true)
const tokenSearchResponse = await searchTokenInAssetsList(value)
setIsLoading(false)
if(tokenSearchResponse && tokenSearchResponse.length){
...
}
```